src/openfermion/ops/_code_operator.py

```diff
@@ -0,0 +1,298 @@
+import copy
+
+import numpy
+import scipy
+import scipy.sparse
+
+from openfermion.ops import SymbolicBinary
+
+
+def _shift_decoder(decode, shift_constant):
+    """ Shifts the indices of a decoder by a constant.
+    
+    Args:
+        decode (list of SymbolicBinary): a decoder
+        shift_constant (int): the qubit index that corresponds to the offset.
+
+    Returns (list of SymbolicBinary):  shifted decoder
+    """
+    decode_shifted = []
+    for entry in decode:
+        tmp_entry = copy.deepcopy(entry)
+        tmp_entry._shift(shift_constant)
```

Comment:
The fact that you are calling ._shift() externally on the class SymbolicBinary (as opposed to calling some public method which is using ._shift() as part of the internal implementation) makes me think that this ._shift() method should also be a public method, i.e. .shift().